File: rust/src/2864.maximum-odd-binary-number/solution.rs
// Created by Jones at 2024/03/01 11:16
// leetgo: 1.4.1
// https://leetcode.com/problems/maximum-odd-binary-number/

use anyhow::Result;
use leetgo_rs::*;

struct Solution;

// @lc code=begin
use std::cmp::Reverse;
use std::collections::*;
use std::mem::swap;

impl Solution {
    pub fn maximum_odd_binary_number(s: String) -> String {
        let n = s.len();
        let cnt = s.bytes().filter(|ch| *ch == b'1').count();
        format!("{}{}1", "1".repeat(cnt - 1), "0".repeat(n - cnt))
    }
}

// @lc code=end

fn main() -> Result<()> {
    let s: String = deserialize(&read_line()?)?;
    let ans: String = Solution::maximum_odd_binary_number(s).into();

    println!("\noutput: {}", serialize(ans)?);
    Ok(())
}

File: python/2864.maximum-odd-binary-number/solution.py
# Created by Jones at 2024/03/13 13:01
# leetgo: 1.4.2
# https://leetcode.cn/problems/maximum-odd-binary-number/

from typing import *
from leetgo_py import *

from bisect import bisect_left, bisect_right
from collections import Counter, defaultdict, deque
from copy import deepcopy
from functools import cache, cmp_to_key, lru_cache, reduce
from heapq import heapify, heappop, heappush, heappushpop, heapreplace
from itertools import accumulate, chain, count, pairwise, zip_longest
from math import ceil, comb, floor, gcd, inf, isqrt, log2, perm, sqrt
from operator import xor
from pprint import pprint
from string import ascii_lowercase
from typing import List, Optional

# @lc code=begin


class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        cnt = s.count("1")
        return "1" * (cnt - 1) + "0" * (len(s) - cnt) + "1"


# @lc code=end

if __name__ == "__main__":
    s: str = deserialize("str", read_line())
    ans = Solution().maximumOddBinaryNumber(s)
    print("\noutput:", serialize(ans, "string"))
